src/js/media-theme-element.js

```diff
@@ -27,11 +29,34 @@ export class MediaThemeElement extends window.HTMLElement {
     super();
     this.renderRoot = this.attachShadow({ mode: 'open' });
 
-    const observer = new MutationObserver(() => this.render());
+    const observer = new MutationObserver((mutationList) => {
+      if (mutationList.some((mutation) => {
+        const target = /** @type HTMLElement */ (mutation.target);
+
+        // Render on each attribute change of the `<media-theme(-x)>` element.
+        if (this.hasAttribute(mutation.attributeName)) return true;
+
+        // Only check `<media-controller>`'s attributes below.
+        if (target.localName !== 'media-controller') return false;
```

Comment:
suggestion(non-blocking): As long as we're not worried about the module ref, consider `target instanceof MediaController` instead